### PR TITLE
Feat 628 improve datadesc upgrader

### LINKED ISSUE #628: DataDescriptionUpgrader improvements

### PR DESCRIPTION
**Pull request by Sun-flow**
closes #628 

Updates to `DataDescriptionUpgrader`, `DataDescription`, and `ModalityUpgrader`.

`map_modality` function added to `data_description_upgrade.py` to convert outdated modalities to modern ones.

similar lookup added to `DataDescription` for `DataLevel` validation.

Error handling for funding if funding does not exist.

### CODE REVIEW

File: src/aind_data_schema/schema_upgrade/data_description_upgrade.py [hunk 1]
```diff
@@ -94,7 +119,13 @@ def upgrade(self, **kwargs) -> AindModel:
             self._get_or_default(self.old_model, "institution", kwargs)
         )
         funding_source = self._get_or_default(self.old_model, "funding_source", kwargs)
-        funding_source = [FundingUpgrade.upgrade_funding(funding) for funding in funding_source]
+        
+        if funding_source is not None:
```
Comment (dyf) [inline src/aind_data_schema/schema_upgrade/data_description_upgrade.py:123]: let's move this logic all into a new `FundingUpgrade.upgrade_funding_source`. this line would turn into:

`funding_source = FundingUpgrade.upgrade_funding_source(funding_source)`.

File: tests/test_schema_upgrade.py [hunk 2]
```diff
@@ -329,6 +329,31 @@ def test_modality_upgrade(self):
...
+            'name': 'SmartSPIM_623711_2022-10-27_16-48-54_stitched_2022-11-01_16-01-12',
+            'institution': 'AIND',
+            'funding_source': None,
+            'data_level': 'derived data',
+            'group': None,
+            'project_name': None,
+            'project_id': None,
+            'restrictions': None,
+            'modality': 'SmartSPIM',
+            'platform': Platform('SMARTSPIM'),
```
Comment (dyf) [inline tests/test_schema_upgrade.py:349]:
```
`platform`: Platform.SMARTSPIM
```
 is simpler, I think?

**Review by dyf — CHANGES_REQUESTED** [review]
couple small things, but otherwise looks okay

**Comment by jtyoung84** [comment]
@Sun-flow I incorporated this into the V2 upgrade. Can you verify that it's there and whether this can be closed?

**Pull request closed without merging**